flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -145,6 +145,12 @@
 
 	private static final long serialVersionUID = 1L;
 
+	/**
+	 * Number of characters to truncate the taskName to for the Kafka transactionalId.
+	 * The maximum this can possibly be set to is 32,767 - (length of operatorUniqueId).
+	 */
+	private static final short maxTaskNameSize = 10_000;
```

Comment:
Wasn't sure what to go with as a value here but using the max characters seems unnecessarily expensive and probably doesn't provide a ton of signal. 

If others agree, I thinking moving to something even lower may even be better than 10k given I believe the transactionId is sent over the network to the broker which is a lot of wasted network overhead.

Reply:
Yes, I think even a string length of 10k is excessive. Let's maybe limit it to 1k?